docs/agents/tools.md

```diff
@@ -26,6 +26,47 @@ main_agent = AgentConfig(
 )
 ```
 
+### Agents-as-tools
+
+As any callable can be used as a tool, you can wrap calling an agent inside a function and pass it as a tool, as in the following example:
+
+```python
+import asyncio
+from any_agent import AgentConfig, AgentFramework, AnyAgent, AgentTrace
+from any_agent.tools import search_web
+
+async def main():
+    google_agent = AnyAgent.create(
+        "google",
+        AgentConfig(
+            name="google_expert",
+            model_id="gpt-4.1-mini",
+            description="An agent that can answer questions about the Google Agents Development Kit (ADK).",
+            tools=[search_web]
+        )
+    )
+
+    # Note that an agent with structured output
+    async def google_agent_as_tool(query: str) -> str:
+        out = await google_agent.run_async(prompt=query)
+        return str(out.final_output)
+
+    google_agent_as_tool.__doc__ = google_agent.config.description
+
+    main_agent_cfg = AgentConfig(
+        instructions="Use the available tools to obtain additional information to answer the query.",
+        tools=[
+            google_agent_as_tool
+        ],
+        model_id="gpt-4.1-nano",
+    )
```

Comment:
Please run this code to make sure it works, because since it's wrapped in a main() that is never called, it will be skipped by our docs tests.

Maybe there should be an asyncio.run(main()) to call the code?

Reply:
There was, previously, but David prefers to have a shorter example.

The code indeed works (but it's not pushed... maybe we could have an examples dir, or the like).

Reply:
> The code indeed works (but it's not pushed... maybe we could have an examples dir, or the like).

The thing is that is under `async def main`, so it never runs. If you put it outside and use sync API, it would run as part of the docs tests, otherwise we just need to add what Nathan suggested. It doesn't matter how short/long the example is and it is only about definitions and not executing the agent